src/lib/zkapps_examples/calls/zkapps_calls.ml

```diff
@@ -0,0 +1,354 @@
+open Core_kernel
+open Snark_params.Tick.Run
+open Signature_lib
+open Mina_base
+
+(** The type underlying the opaque call data digest field used by the call
+    rules.
+*)
+module Call_data = struct
+  (** The type of inputs that the zkApp rules receive when called by another
+      zkApp.
+  *)
+  module Input = struct
+    module Constant = struct
+      type t = { old_state : Field.Constant.t } [@@deriving hlist]
+
+      let to_ro_input { old_state } =
+        Random_oracle_input.Chunked.field_elements [| old_state |]
+    end
+
+    module Circuit = struct
+      type t = { old_state : Field.t } [@@deriving hlist]
+
+      let to_ro_input { old_state } =
+        Random_oracle_input.Chunked.field_elements [| old_state |]
+    end
+
+    let typ =
+      Typ.of_hlistable ~value_to_hlist:Constant.to_hlist
+        ~value_of_hlist:Constant.of_hlist ~var_to_hlist:Circuit.to_hlist
+        ~var_of_hlist:Circuit.of_hlist [ Field.typ ]
+  end
+
+  (** The type of outputs that the zkApp rules receive when called by another
+      zkApp.
+  *)
+  module Output = struct
+    module Constant = struct
+      type t =
+        { blinding_value : Field.Constant.t; new_state : Field.Constant.t }
```

Comment:
It makes the value 'secret', so that the communication between the zkApps can contain private data without anybody being able to figure it out on-chain. I'll put it in a comment.